### DISCUSSION TITLE
How can I recognize history type ("WebHistory" or "WebHashHistory") of a router instance?

### DISCUSSION THREAD

**Discussion by cssmagic** [vuejs/router#1858] [Help and Questions] [answered]
I'm writing a utility which needs to perform different actions depending on the type of history.

I logged a router instance to console, digged into `.options.history` but found nothing about its type.

Do we have any way to recognize that?

**Comment by posva** [comment, marked as answer]
There is no way to recognize this as history is instantiated outside of the router now. What you could do is let the user pass what version of the history they are using or take control over router creation to pass the options you want.
